tests/from_msa/test_prg_builder.py

```diff
@@ -0,0 +1,146 @@
+from pathlib import Path
+from unittest import TestCase
+
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+from make_prg.from_msa.prg_builder import PrgBuilder
+from tests.from_msa import make_alignment, MSA
+
+this_dir = Path(__file__).resolve().parent
+data_dir = this_dir.parent / "data" / "make_prg_from_msa"
+
+
+class TestConsensusString(TestCase):
+    def test_all_match(self):
+        alignment = make_alignment(["AATTA", "AATTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "AATTA")
+
+    def test_mixed_match_nonmatch(self):
+        alignment = make_alignment(["AAGTA", "CATTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "*A*TA")
+
+    def test_indel_nonmatch(self):
+        alignment = make_alignment(["AAAA", "A--A"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "A**A")
+
+    def test_IUPACAmbiguous_nonmatch(self):
+        alignment = make_alignment(["RYA", "RTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "**A")
+
+    def test_N_special_treatment(self):
+        """
+        i)A and N at pos 2 are different, but still consensus
+        ii)N and N at pos 0 are same, but not consensus"""
+        alignment = make_alignment(["NTN", "NTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "*TA")
+
+    def test_all_gap_nonmatch(self):
+        alignment = make_alignment(["A--A", "A--A"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "A**A")
```

Comment:
Why would this not be `AA`?

Reply:
It becomes that eventually- at this stage, we're just marking whether each position is match or non-match. If you have all `N` or all `-`, that doesn't count as match: you can't actually use that as a non-variant bit of the PRG